client/lib/plans/index.js

```diff
@@ -143,6 +149,36 @@ export function filterPlansBySiteAndProps( plans, site, hideFreePlan, intervalTy
 	} );
 }
 
+export function getMonthlyPlanByYearly( plan ) {
+	switch ( plan ) {
+		case PLAN_JETPACK_PREMIUM:
+			return PLAN_JETPACK_PREMIUM_MONTHLY;
+		case PLAN_JETPACK_BUSINESS:
+			return PLAN_JETPACK_BUSINESS_MONTHLY;
+		case PLAN_JETPACK_PERSONAL:
+			return PLAN_JETPACK_PERSONAL_MONTHLY;
+		default:
+			return '';
+	}
+}
+
+/**
+ * Returns true if plans match regardeless of their interval.
+ * For example (fake plans):
+ *     isPlanMatch( 'proYearly', 'proYearly' ) => true
+ *     isPlanMatch( 'proYearly', 'proMonthly' ) => true
+ *     isPlanMatch( 'proYearly', 'personalYearly' ) => false
+ *
+ * @param  {String}  planSlugA One of the plan slugs to compare
+ * @param  {String}  planSlugB One of the plan slugs to compare
+ * @return {Boolean}           Whether the plans match
+ */
+export function isPlanMatch( planSlugA, planSlugB ) {
+	return planSlugA === planSlugB ||
```

Comment:
I'm late to this, so feel free to ignore, but another way to do it is have a `convertToYearlyPlan()` (or monthly) func that returns yearly plans unchanged, then this becomes simply:

`return convertToYearly( planSlugA) === convertToYearly( planSlugB);`